src/GovUk.Education.ExploreEducationStatistics.Common.Tests/Utils/CompressionUtilsTests.cs

```diff
@@ -0,0 +1,173 @@
+﻿#nullable enable
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Text;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Utils;
+using Xunit;
+using ZstdSharp;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Tests.Utils;
+
+public class CompressionUtilsTests
+{
+    private const string Content = "Test content";
+
+    [Fact]
+    public async Task CompressToStream_UnsupportedContentEncoding()
+    {
+        await Assert.ThrowsAsync<NotSupportedException>(async () =>
+            await CompressionUtils.CompressToStream(
+                stream: Content.ToStream(),
+                targetStream: new MemoryStream(),
+                contentEncoding: "other"));
+    }
+
+    [Fact]
+    public async Task CompressToStream_Gzip()
+    {
+        // Compress some content to a stream using the method under test
+        await using var compressedStream = new MemoryStream();
+        await CompressionUtils.CompressToStream(
+            stream: Content.ToStream(),
+            targetStream: compressedStream,
+            contentEncoding: ContentEncodings.Gzip);
+
+        Assert.Equal(0, compressedStream.Position);
+        Assert.True(compressedStream.Length > 0);
```

Comment:
Worth checking that the compressedStream.Length is less than Content.ToStream().Length at this point?

Reply:
Or at least that it's not still equal to the original Content Stream at this point?

Reply:
I think, depending on content, the compressed stream can be longer than the original content stream. I've gone with checking that it's not equal to the original content stream.